tests_e2e/test_latest_build.py

```diff
@@ -0,0 +1,51 @@
+"""
+This pytest test opens the ../dist/basic_bot-0.1.0.tar.gz file and
+asserts the following:
+
+- The file exists
+
+- The uncompressed file has the following files or directories:
+    src/basic_bot/__init__.py
+    src/basic_bot/create.py
+    src/basic_bot/created_files
+    src/basic_bot/created_files/webapp
+
+- The uncompressed file does not contain anyof  the following files or
+directories recursively:
+    __pycache__
+    node_modules
+    package-lock.json
+
+"""
+
+import tarfile
+from pathlib import Path
+
+DIST_FILE = Path("./dist/basic_bot-0.1.0.tar.gz")
+REQUIRED_FILES = [
+    "src/basic_bot/__init__.py",
+    "src/basic_bot/create.py",
+    "src/basic_bot/created_files",
+    "src/basic_bot/created_files/webapp",
+]
+EXCLUDED_PATTERNS = ["__pycache__", "node_modules", "package-lock.json"]
```

Comment:
I tested the test manually by first adding a bogus file to `REQUIRED_FILES` and separately adding `package.json` which should exist to `EXCLUDED_PATTERNS`.